perfkitbenchmarker/linux_benchmarks/cloudsuite_data_caching_benchmark.py

```
@@ -0,0 +1,144 @@
+
+flags.DEFINE_string('cloudsuite_data_caching_memcached_flags',
+                    '-t 1 -m 2048 -n 550',
+                    'Flags to be given to memcached.')
+flags.DEFINE_string('cloudsuite_data_caching_rps',
+                    '18000',
+                    'Number of requests per second.')
+FLAGS = flags.FLAGS
+
+BENCHMARK_NAME = 'cloudsuite_data_caching'
+BENCHMARK_CONFIG = """
+cloudsuite_data_caching:
+  description: >
+    Runs Cloudsuite3.0 Data Caching benchmark.
+  vm_groups:
+    server:
+      vm_spec: *default_single_core
+    client:
+      vm_spec: *default_single_core
+"""
+
+
+def GetConfig(user_config):
+  return configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+
+
+def Prepare(benchmark_spec):
+  """Install docker. Pull the required images from DockerHub. Create datasets.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  vms = benchmark_spec.vms
+  server_vm = benchmark_spec.vm_groups['server'][0]
+  client_vm = benchmark_spec.vm_groups['client'][0]
+
+# Make sure docker is installed on all VMs.
+  for vm in vms:
+    if not docker.IsInstalled(vm):
+      vm.Install('docker')
+
+    # Prepare and start the server VM.
+  server_vm.RemoteCommand('sudo docker pull cloudsuite/data-caching:server')
+  server_vm.RemoteCommand("echo '%s    dc-client' | sudo tee -a /etc/hosts >"
+                          " /dev/null" % client_vm.internal_ip)
+  server_vm.RemoteCommand('sudo docker run --name dc-server --net host -d '
+                          'cloudsuite/data-caching:server %s' %
+                          FLAGS.cloudsuite_data_caching_memcached_flags)
+
+# Prepare the client.
+  client_vm.RemoteCommand('sudo docker pull cloudsuite/data-caching:client')
+  client_vm.RemoteCommand("echo '%s    dc-server' | sudo tee -a /etc/hosts >"
+                          " /dev/null" % server_vm.internal_ip)
+
+
+def _ParseOutput(output_str):
```

My biggest concern here is that if anything changes in the output format, this method won't complain. It will just silently use the wrong numbers, and the users may not notice. If you want to check this method in as-is, please take other precautions to make the benchmark as stable as possible, for example by pulling a specific tag/digest in the Prepare docker commands.